text/3621-extended_hrtb.md

```diff
@@ -0,0 +1,93 @@
+- Feature Name: `extended_hrtb`
+- Start Date: 2022-05-08
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/3621)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This feature add a way to bound universal quantification of HTRBs to allow more APIs.
+
+# Motivation
+[motivation]: #motivation
+
+The HRTB construct has allowed us to talk about closures that are generic over lifetimes. However, initially this has not included a way to restrict these generic lifetimes, thus, in practice, it turned out to be useful in only a handful of scenarios.
+
+The goal of the proposal is to allow bounded universal quantification over lifetimes.
```

Comment:
This explanation is honestly quite confusing to me, and could probably be better worded directly using "outlives" statements like:

Currently, the existing higher-rank trait bounds functionality allows generalising bounds that outlive a given lifetime, but do not allow generalising bounds that are outlived by other lifetimes. This limits the situations that are useful for these bounds, since lifetimes can always be changed to live shorter, but not the other way around.

(definitely open to more rewording there, but at least using that explanation makes things a bit more clear to me)

Reply:
How it feels now?

Reply:
It's good; thank you!